src/smtsolvers/CoreSMTSolver.cc

```diff
@@ -1897,3 +1897,28 @@ void CoreSMTSolver::fillBooleanVars(ModelBuilder &modelBuilder) {
         modelBuilder.addVarValue(atom, val);
     }
 }
+
+vec<CRef> CoreSMTSolver::getUnsatCoreClauses() const {
+    assert(resolutionProof);
+    vec<CRef> unsatCore;
+    auto const & derivations = resolutionProof->getProof();
+    std::vector<CRef> stack {CRef_Undef};
+    std::unordered_set<CRef> processed;
+    do {
```

Comment:
Maybe I would rather make it a while loop than a do-while. It is more clear and the optimization is negligible..